include/CL/sycl/detail/small_array.hpp

```diff
@@ -129,6 +129,41 @@ struct small_array : std::array<BasicType, Dims>,
     return (*this)[2];
   }
 
+  /** An accessor to the third variable of a small array
+  */
+  BasicType& w(){
+    static_assert(Dims >= 4, "can't access to small_array[3] if Dims < 4");
+    return (*this)[3];
+  }
+
+#define DECLARE_S(x)							\
```

Comment:
I think it is safer to use macro names with some prefixes such as `TRISYCL_DECLARE_S` to avoid weird clashes with some user-defined macros...